### PR TITLE
Robotics Expansion: RIG-ged and Ready

### PR DESCRIPTION
**Pull request by Mechoid**
- Move Rig modules to more granular files.
- Add RIG components to robotics mechfab, plus incredibly basic belt-voidsuit RIG.
- Add Graphite, made by compressing coal, presently only used heavily in RIG components.

### CODE REVIEW

File: code/modules/clothing/spacesuits/rig/suits/robotics.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//Mining suit
+/obj/item/weapon/rig/robotics
+	name = "advanced suit control belt"
+	suit_type = "advanced suit"
+	desc = "A lightweight suit combining the utility of a RIG with the wearability of a voidsuit."
+	icon_state = "void_explorer2"
+	slot_flags = SLOT_BELT
```
Comment (Arokha) [inline code/modules/clothing/spacesuits/rig/suits/robotics.dm]: This is awesome and I want it right now. That having been said, I need to point out `/mob/living/carbon/human/get_breath_from_internal` currently checks specifically the `human.back` var for finding a rig to breathe out of. You'll need to update that. I notice there's a var on humans called `wearing_rig` that can likely be used, as rigs set it to themselves when they're worn.

**Comment by Mechoid** [comment]
Why was bikeshed there

**Comment by Atermonera** [comment]
Because y'all are giving me too many huge prs to review :V

File: code/modules/clothing/spacesuits/rig/modules/specific/voice.dm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/obj/item/rig_module/voice/installed()
+	..()
+	holder.speech = src
+
+/obj/item/rig_module/voice/engage()
+
+	if(!..())
+		return 0
+
+	var/choice= input("Would you like to toggle the synthesiser or set the name?") as null|anything in list("Enable","Disable","Set Name")
```
Comment (Atermonera) [inline code/modules/clothing/spacesuits/rig/modules/specific/voice.dm]: input from whom?

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 3]
```diff
@@ -418,29 +428,14 @@
 	interface_name = "mop projector"
```
Comment (Atermonera) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]: How is this mop different from the mop in `mounted_guns.dm`?

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 4]
```diff
@@ -117,8 +129,6 @@
 		device.afterattack(target,holder.wearer,1)
 	return 1
 
-
-
 /obj/item/rig_module/chem_dispenser
 	name = "mounted chemical dispenser"
```
Comment (Atermonera) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]: How is this different from the chem dispenser defined in `chem_dispenser.dm`?

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 5]
```diff
@@ -418,29 +428,14 @@
...
-	if(!..())
-		return 0
-
-	if(!target)
-		gun.attack_self(holder.wearer)
-		return 1
-
-	gun.Fire(target,holder.wearer)
-	return 1
+	gun_type = /obj/item/weapon/gun/energy/temperature/mounted
```
Comment (Atermonera) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]: If it uses a gun, shouldn't it go with the other mounted gun modules?

File: code/modules/clothing/spacesuits/rig/modules/specific/ai_container.dm [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+				eject_ai()
+		else
+			// You're using an empty card on an empty suit, idiot.
+			if(!target_ai)
+				return 0
+			integrate_ai(input_device,user)
+		return 1
+
+	// Okay, it wasn't a terminal being touched, check for all the simple insertions.
+	if(input_device.type in list(/obj/item/device/paicard, /obj/item/device/mmi, /obj/item/device/mmi/digital/posibrain))
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/specific/ai_container.dm]: Those and posibrains are actually covered already by device/mmi

File: code/modules/clothing/spacesuits/rig/modules/specific/grenade_launcher.dm [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	var/fire_force = 30
+	var/fire_distance = 10
+
+	charges = list(
+		list("flashbang",   "flashbang",   /obj/item/weapon/grenade/flashbang,  3),
+		list("smoke bomb",  "smoke bomb",  /obj/item/weapon/grenade/smokebomb,  3),
+		list("EMP grenade", "EMP grenade", /obj/item/weapon/grenade/empgrenade, 3),
+		)
+
+/obj/item/rig_module/grenade_launcher/accepts_item(var/obj/item/input_device, var/mob/living/user)
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/specific/grenade_launcher.dm]: They do not accept chem grenades

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 5 above)
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm]: Theoretically, though its main 'purpose' is the mop.

File: code/modules/clothing/spacesuits/rig/modules/specific/mounted_gun.dm [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+/obj/item/rig_module/mounted/engage(atom/target)
+
+	if(!..())
+		return 0
+
+	if(!target)
+		gun.attack_self(holder.wearer)
+		return
+
+	gun.Fire(target,holder.wearer)
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/specific/mounted_gun.dm]: The Gun is not set to the sword or mop. These are all modules that already exist, I've just moved them into their own files.

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 4 above)
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm]: These are the de-selected files.

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 3 above)
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm]: These are the de-selected files.

File: code/modules/clothing/spacesuits/rig/modules/specific/voice.dm (hunk 2 above)
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/specific/voice.dm]: I assume usr as the module works.

File: code/modules/clothing/spacesuits/rig/modules/specific/mounted_gun.dm [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+	active_power_cost = 10
+	passive_power_cost = 0
+
+	gun_type = /obj/item/weapon/gun/energy/crossbow/ninja
+
+/obj/item/rig_module/mounted/energy_blade/process()
+
+	if(holder && holder.wearer)
+		if(!(locate(/obj/item/weapon/melee/energy/blade) in holder.wearer))
+			deactivate()
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/clothing/spacesuits/rig/modules/specific/mounted_gun.dm]: The suit deactivates modules when it runs out of power.

**Pull request merged**